src/TestInfrastructure/Helpers/ProcessScope.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Helpers
+{
+    /// <summary>
+    /// Ensure the process is terminated when it goes out of scope.
+    /// Usage:
+    ///   using var processScope = new ProcessScope(process);
+    /// </summary>
+    /// <remarks>Disposing a process does not terminate it so we can't rely on
+    /// "using var process = ..." to clean up properly.
+    /// Instead, we'll try to fetch the process by id then "Kill" it.
+    /// </remarks>
+    public sealed class ProcessScope : IDisposable
+    {
+        private readonly int id;
+        private readonly string mainModuleFilName;
+        private readonly string processName;
+
+        public Process Process { get; }
+
+        public ProcessScope(Process process)
+        {
+            Process = process ?? throw new ArgumentNullException(nameof(process));
+
+            // Cache the properties we need now as we won't be able to access them
+            // later if the process has been disposed at the point we want to clean up.
+            id = process.Id;
+
+            (mainModuleFilName, processName) = TryGetProcessDetails(process);
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                var runningProcess = Process.GetProcessById(id);
```

Comment:
Why not dispose the existing Process property?

Reply:
Disposing won't help, we need to call Kill.

Could do, but the call to Kill would fail if the process has been disposed (as would calls to any other members). So we'd have to make that Kill call inside a separate try-catch block, then fall back GetProcessById if that didn't work.

The GetProcessById version should work regardless of whether the original Process has been disposed or not.